client/js/contextMenuFactory.js

```diff
@@ -125,6 +125,74 @@ function addKickItem() {
 	});
 }
 
+function addOpItem() {
+	function op(itemData) {
+		socket.emit("input", {
+			target: $("#chat").data("id"),
+			text: "/op " + itemData,
+		});
+	}
+
+	addContextMenuItem({
+		check: (target) => utils.hasRoleInChannel(target.closest(".chan"), ["op"]) && utils.hasRoleInChannel(target.closest(".chan"), ["op"], target.data("name")) === false,
+		className: "action-op",
+		displayName: "Give op (+o)",
```

Comment:
I'm really not a fan of "op". We use "Operators" in the user list menu and I find it more appropriate.
You'll need to change style.css L1896 from 160px to 180px.